cmd/algorand-indexer/daemon.go

```diff
@@ -49,50 +52,131 @@ var (
 	defaultApplicationsLimit  uint32
 	enableAllParameters       bool
 	indexerDataDir            string
+	cpuProfile                string
+	pidFilePath               string
+	configFile                string
 )
 
 var daemonCmd = &cobra.Command{
 	Use:   "daemon",
 	Short: "run indexer daemon",
 	Long:  "run indexer daemon. Serve api on HTTP.",
-	//Args:
 	Run: func(cmd *cobra.Command, args []string) {
 		var err error
 		config.BindFlags(cmd)
 		err = configureLogger()
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "failed to configure logger: %v", err)
-			os.Exit(1)
+			panic(exit{1})
+		}
+
+		if indexerDataDir == "" {
+			fmt.Fprint(os.Stderr, "indexer data directory was not provided")
+			panic(exit{1})
+		}
+
+		// Create the directory if it doesn't exist
+		if _, err := os.Stat(indexerDataDir); os.IsNotExist(err) {
+			err := os.Mkdir(indexerDataDir, 0755)
+			maybeFail(err, "indexer data directory error, %v", err)
+		}
+
+		// Detect the various auto-loading configs from data directory
+		indexerConfigFound := util.FileExists(filepath.Join(indexerDataDir, autoLoadIndexerConfigName))
+		paramConfigFound := util.FileExists(filepath.Join(indexerDataDir, autoLoadParameterConfigName))
+
+		// If we auto-loaded configs but a user supplied them as well, we have an error
+		if indexerConfigFound {
+			if configFile != "" {
+				logger.Errorf(
+					"indexer configuration was found in data directory (%s) as well as supplied via command line.  Only provide one.",
+					filepath.Join(indexerDataDir, autoLoadIndexerConfigName))
+				panic(exit{1})
+			} else {
+				// No config file supplied via command line, auto-load it
+				configs, err := os.Open(configFile)
+				if err != nil {
+					maybeFail(err, "%v", err)
+				}
+				defer configs.Close()
+				err = viper.ReadConfig(configs)
+				if err != nil {
+					maybeFail(err, "invalid config file (%s): %v", viper.ConfigFileUsed(), err)
+				}
+				fmt.Printf("Using configuration file: %s\n", configFile)
+			}
```

Comment:
nit: the linter would have made this suggestion if we were using normal error handling
```suggestion
			}
			// No config file supplied via command line, auto-load it
			configs, err := os.Open(configFile)
			if err != nil {
				maybeFail(err, "%v", err)
			}
			defer configs.Close()
			err = viper.ReadConfig(configs)
			if err != nil {
				maybeFail(err, "invalid config file (%s): %v", viper.ConfigFileUsed(), err)
			}
			fmt.Printf("Using configuration file: %s\n", configFile)
```